File: 3455-minimum-length-of-string-after-operations/3455-minimum-length-of-string-after-operations.py
class Solution:
    def minimumLength(self, s: str) -> int:
        hash_map = {}

        for char in s:
            if char in hash_map:
              hash_map[char] += 1
            else:
                hash_map[char] = 1 
        values = list(hash_map.values())

        for i in range(len(values)):
            if values[i] >= 3:
                while values[i] >= 3:
                   values[i] -= 2
        return sum(values)
        
                

File: 3455-minimum-length-of-string-after-operations/3455-minimum-length-of-string-after-operations.js
/**
 * @param {string} s
 * @return {number}
 */
var minimumLength = function(s) {
    let hash_map = new Map()

    for(let char of s){
        if(hash_map.has(char)){
            hash_map.set(char,hash_map.get(char) + 1)
        }else{
            hash_map.set(char,1)
        }
    }
     
     for(let [key,value] of hash_map){
        if(value >= 3){
            while (value >= 3) {
             value -= 2;
            }
            hash_map.set(key, value)
        } 
     }
     const values = [...hash_map.values()]

     return values.reduce((acc,cur) => cur + acc)
};